### PR TITLE
Format notebooks with Ruff

### PR DESCRIPTION
**Pull request by frankharkins**
Follow-up to #2197. Draft until that PR is merged, then I'll rebase this.

Also a partial fix for #1806 depending if we want to also lint code examples in MDX.

### CODE REVIEW

File: docs/ruff.toml
Comment (frankharkins) [inline docs/ruff.toml:1]: This seems to be the longest length that doesn't scroll horizontally on the website. I tested by inspect element then adding characters until the box scrolled on Firefox.

**Review by Eric-Arellano — APPROVED** [review]
❤️

I forgot to mention, can you please check that the README is up to date?

**Comment by frankharkins** [comment]
Gonna wait to merge this until after #2139 is merged to avoid further disruption.

**Review by Eric-Arellano — APPROVED** [review]

**Pull request merged**
